test/jdk/tools/sincechecker/SinceChecker.java

```diff
@@ -0,0 +1,769 @@
+                return true;
+            }
+            el = el.getEnclosingElement();
+        }
+
+        return LEGACY_PREVIEW_METHODS.containsKey(currentVersion)
+                && LEGACY_PREVIEW_METHODS.get(currentVersion).contains(uniqueId);
+    }
+
+    private void testThisModule(String moduleName) throws Exception {
+        Path home = Paths.get(System.getProperty("java.home"));
+        Path srcZip = home.resolve("lib").resolve("src.zip");
+        if (Files.notExists(srcZip)) {
+            //possibly running over an exploded JDK build, attempt to find a
+            //co-located full JDK image with src.zip:
+            Path testJdk = Paths.get(System.getProperty("test.jdk"));
+            srcZip = testJdk.getParent().resolve("images").resolve("jdk").resolve("lib").resolve("src.zip");
+        }
+        File f = new File(srcZip.toUri());
+        if (!f.exists() && !f.isDirectory()) {
+            throw new SkippedException("Skipping Test because src.zip wasn't found");
+        }
+        if (Files.isReadable(srcZip)) {
+            URI uri = URI.create("jar:" + srcZip.toUri());
+            try (FileSystem zipFO = FileSystems.newFileSystem(uri, Collections.emptyMap())) {
+                Path root = zipFO.getRootDirectories().iterator().next();
+                Path packagePath = root.resolve(moduleName);
+                try (StandardJavaFileManager fm =
+                             tool.getStandardFileManager(null, null, null)) {
+                    JavacTask ct = (JavacTask) tool.getTask(null,
+                            fm,
+                            null,
+                            List.of("--add-modules", moduleName, "-d", "."),
+                            null,
+                            Collections.singletonList(SimpleJavaFileObject.forSource(URI.create("myfo:/Test.java"), "")));
+                    ct.analyze();
+                    Elements elements = ct.getElements();
+                    elements.getModuleElement("java.base");
+                    try (EffectiveSourceSinceHelper javadocHelper = EffectiveSourceSinceHelper.create(ct, List.of(root), this)) {
+                        processModuleCheck(elements.getModuleElement(moduleName), ct, packagePath, javadocHelper);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        wrongTagsList.add("Initiating javadocHelperFailed " + e.getMessage());
+                    }
+                    if (!wrongTagsList.isEmpty()) {
+                        throw new Exception(wrongTagsList.toString());
+                    }
+                }
+            }
+        }
+    }
+
+    private void processModuleCheck(ModuleElement moduleElement, JavacTask ct, Path packagePath, EffectiveSourceSinceHelper javadocHelper) {
+        if (moduleElement == null) {
+            wrongTagsList.add("Module element: was null because `elements.getModuleElement(moduleName)` returns null." +
```

Comment:
This doesn't look like a "Wrong tag" so much as a general error message.

Consider this as a different paradigm -- instead of saving up strings in `wrongTagsList`, consider having a method to report (and count) error messages as you find them:

```
    private int errorCount;
    void error(String message) {
        System.err.println(message);
        errorCount++;
   }
```

then at the end of `testThisModule`, just check if `errorCount>0` to decide whether to throw the exception.

Note, in this model, you need not have a `\n` at the end of every string argument, the way you do for add strings added into `wrongTagsList`.

Reply:
Changed it and removed '\n'